lib/model/folder.go

```diff
@@ -500,16 +507,28 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 			}
 			batch.append(fi)
 		}
-	}
-
-	// Schedule a pull after scanning, but only if we actually detected any
-	// changes.
-	changes := 0
-	defer func() {
-		if changes > 0 {
-			f.SchedulePull()
+	case config.FolderTypeReceiveEncrypted:
+		batchAppend = func(fi protocol.FileInfo, _ *db.Snapshot) {
+			// This is a "virtual" parent directory of encrypted files.
+			// We don't track it, but check if anything still exists
+			// within and delete it otherwise.
+			if fi.IsDirectory() && protocol.IsEncryptedParent(fi.Name) {
+				if names, err := mtimefs.DirNames(fi.Name); err == nil && len(names) == 0 {
+					mtimefs.Remove(fi.Name)
+				}
+				changes--
+				return
+			}
```

Comment:
The purpose is to have files in many dirs, not just all of them in one dir (the root dir). That's achieved by splitting using the first few filename chars as a parent dir. These dirs are ignored, i.e. they don't get a file info on scanning and we just create them in the puller and remove in scanner if empty. Seems like a functional and simple hack to me.

Reply:
What happens to directories that we announce on the plaintext side?

Reply:
They are treated the same way, i.e. also using first few chars of the encrypted name as parent dirs. The "final" component then is just a dir instead of a file (symlinks are represented as dirs).